server.go

```diff
@@ -125,8 +154,61 @@ func getVault(subscriptionID string, vaultName string) (vaultUrl *string, err er
 	return vault.Properties.VaultURI, nil
 }
 
-func createKey(keyClient kv.ManagementClient, vaultUrl string, keyName string) (*string, error) {
+func createKey(keyClient kv.ManagementClient, vaultUrl string, keyName string, providerVaultName string, subscriptionID string) (*string, error) {
 	fmt.Println("Key not found. Creating a new key...")
+	storageAccountsClient := storagemgmt.NewAccountsClient(subscriptionID)
```

Comment:
We really don't subscription id here. You can get a token for resource by storage account name, and env (cloud name) only. This way we ask the user for less

Reply:
Also doesnt seem like the Azure Go SDK has the AD authentication feature added for storage yet: https://github.com/Azure/azure-sdk-for-go/blob/v18.0.0/storage/client.go#L269-L276

```
func NewBasicClient(accountName, accountKey string) (Client, error) {
	if accountName == StorageEmulatorAccountName {
		return NewEmulatorClient()
	}
	return NewClient(accountName, accountKey, DefaultBaseURL, DefaultAPIVersion, defaultUseHTTPS)
}
```